packages/@lwc/engine/dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -38,3 +49,56 @@ export function deprecatedBuildCustomElementConstructor(
 
     return Ctor.CustomElementConstructor;
 }
+
+export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLElementConstructor {
+    const def = getComponentInternalDef(Ctor);
+
+    // generating the hash table for attributes to avoid duplicate fields and facilitate validation
+    // and false positives in case of inheritance.
+    const attributeToPropMap: Record<string, string> = create(null);
+    for (const propName in def.props) {
```

Comment:
is def guaranteed to exist prior accessing 'props'?

Reply:
Yes.